### PR TITLE
fix(dashboard): fixed the missing in-app editor body message

### PR DESCRIPTION
**Pull request by LetItRock**
### What changed? Why was the change needed?

Fixed the missing body error in the in-app editor. cc @scopsy

### CODE REVIEW

**Review by scopsy — APPROVED** [review]

**Review by BiswaViraj — APPROVED** [review]

File: apps/dashboard/src/components/workflow-editor/steps/in-app/in-app-editor.tsx [hunk 1]
```diff
@@ -44,7 +48,9 @@ export const InAppEditor = ({ uiSchema }: { uiSchema?: UiSchema }) => {
           {body && (
             <>
               {getComponentByType({ component: body.component })}
-              <FormMessage>{`Type {{ for variables, or wrap text in ** for bold.`}</FormMessage>
+              <FormMessagePure
```
Comment (desiprisg) [inline apps/dashboard/src/components/workflow-editor/steps/in-app/in-app-editor.tsx]: Why is this necessary? I'd expect `<FormMessage />` to take care of this.

Comment (LetItRock) [reply apps/dashboard/src/components/workflow-editor/steps/in-app/in-app-editor.tsx]: because it's outside of the `FormField`

**Review by SokratisVidros — APPROVED** [review]

File: apps/dashboard/src/components/workflow-editor/steps/in-app/in-app-editor.tsx (hunk 1 above)
Comment (BiswaViraj) [reply apps/dashboard/src/components/workflow-editor/steps/in-app/in-app-editor.tsx]: Wont it be possible to add it to the `in-app-body` component directly?

Comment (LetItRock) [reply apps/dashboard/src/components/workflow-editor/steps/in-app/in-app-editor.tsx]: It was there, but Dima updated the code and reused the `getComponentByType,` which shouldn't be actually used for onboarding purposes. Maybe I'll take a look into that part.

**Review by SokratisVidros — CHANGES_REQUESTED** [review]
I undo my approval. Check my latest comment.

File: apps/dashboard/src/components/workflow-editor/steps/in-app/in-app-editor.tsx (hunk 1 above)
Comment (SokratisVidros) [reply apps/dashboard/src/components/workflow-editor/steps/in-app/in-app-editor.tsx]: The error message belongs in the `InAppEditor`.

File: apps/dashboard/src/components/auth/customize-inbox-playground.tsx [hunk 2]
```diff
@@ -149,12 +154,62 @@ function ColorPickerSection({ form }: { form: UseFormReturn<InboxPlaygroundFormD
...
+};
+
 function NotificationConfigSection() {
+  const { control } = useFormContext();
+
   return (
     <div className="flex flex-col gap-1 p-1">
-      <div className="flex gap-1">{getComponentByType({ component: UiComponentEnum.IN_APP_SUBJECT })}</div>
-      {getComponentByType({ component: UiComponentEnum.IN_APP_BODY })}
-      {getComponentByType({ component: UiComponentEnum.IN_APP_BUTTON_DROPDOWN })}
```
Comment (LetItRock) [inline apps/dashboard/src/components/auth/customize-inbox-playground.tsx:157]: The `getComponentByType` is the in-app editor specific util that we shouldn't use in any other context. 
So I refactored the code to use the form fields instead.

File: apps/dashboard/src/components/in-app-action-dropdown.tsx [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+import { parseStepVariablesToLiquidVariables } from '@/utils/parseStepVariablesToLiquidVariables';
+import { cn } from '@/utils/ui';
+import { urlTargetTypes } from '@/utils/url';
+import { autocompletion } from '@codemirror/autocomplete';
+import { useStep } from '@/components/workflow-editor/steps/step-provider';
+
+const primaryActionKey = 'primaryAction';
+const secondaryActionKey = 'secondaryAction';
+
+export const InAppActionDropdown = ({ onMenuItemClick }: { onMenuItemClick?: () => void }) => {
```
Comment (LetItRock) [inline apps/dashboard/src/components/in-app-action-dropdown.tsx:37]: This component is now "more" reusable and we use it under the `/onboarding/inbox` route.

**Review by SokratisVidros — APPROVED** [review]
